vendor/engines/secure_rooms/app/models/secure_rooms/event.rb

```diff
@@ -11,6 +11,10 @@ class Event < ActiveRecord::Base
 
     delegate :direction, :secure_room, :ingress?, :egress?, to: :card_reader
 
+    def successful?
```

Comment:
Would it be worth pulling something into the `AccessManager` that breaks out of the loop if a result is not successful? And have all the handlers implement a success predicate? That might be overkill, though.

And just from a nitpicky perspective, I think `success?` is a little more idiomatic.